test/logger_test.rb

```diff
@@ -0,0 +1,85 @@
+require 'test_helper'
+
+describe Lotus::Logger do
+  class IoCapture
+    attr_accessor :stream
+    def write(smt)
+      @stream = smt
+    end
+    def close;end
+    def flush;end
+  end
+
+  #alias STDOUT to custom class to test output
+  STDOUT = IoCapture.new
+
+  before do
+    #clear defined class
+    Object.constants.include?(:TestLogger)
+  end
+
+  it 'like std logger, default log level is info' do
+    class TestLogger1 < Lotus::Logger; end
+    assert_respond_to(TestLogger1.new, :info?)
+  end
+
+  it 'always use STDOUT' do
+    class TestLogger2 < Lotus::Logger; end
+    logger = TestLogger2.new
+    logger.info('foo')
+    assert_match(/foo/, STDOUT.stream)
+  end
+
+  describe '.app_tag' do
+    it 'has app_tag when log' do
+      module App; class TestLogger3 < Lotus::Logger; end; end
+      logger = App::TestLogger3.new
+      logger.info('foo')
+      assert_match(/App/, STDOUT.stream)
+    end
+
+    it 'raise exception if override logger without app_tag' do
+      class TestLogger4 < Lotus::Logger; def app_tag; end; end
+      assert_raises(Exception) { logger = TestLogger4.new.info('anything') }
+    end
+
+    it 'has default app tag when not in any namespace' do
+      class TestLogger5 < Lotus::Logger; end
+      TestLogger5.new.send(:app_tag).must_equal 'Shared'
```

Comment:
Is there reason why we could not invoke `app_tag` but having to go through `send`?